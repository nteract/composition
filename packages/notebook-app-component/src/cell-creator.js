// @flow strict
import * as React from "react";
import { connect } from "react-redux";
import { CodeOcticon, MarkdownOcticon } from "@nteract/octicons";
import { actions } from "@nteract/core";
import type { ContentRef } from "@nteract/core";

type Props = {
  above: boolean,
  createCell: (type: "code" | "markdown") => void
};

type ConnectedProps = {
  above: boolean,
  createCellAppend: (payload: *) => void,
  createCellAbove: (payload: *) => void,
  createCellBelow: (payload: *) => void,
  id?: string,
  contentRef: ContentRef
};

export class PureCellCreator extends React.Component<Props, null> {
  createMarkdownCell = () => {
    this.props.createCell("markdown");
  };

  createCodeCell = () => {
    this.props.createCell("code");
  };

  render() {
    return (
      <div className="creator-hover-mask">
        <div className="creator-hover-region">
          <div className="cell-creator">
            <button
              onClick={this.createMarkdownCell}
              title="create text cell"
              className="add-text-cell"
            >
              <span className="octicon">
                <MarkdownOcticon />
              </span>
            </button>
            <button
              onClick={this.createCodeCell}
              title="create code cell"
              className="add-code-cell"
            >
              <span className="octicon">
                <CodeOcticon />
              </span>
            </button>
          </div>
        </div>
        <style jsx>{`
          .creator-hover-mask {
            display: block;
            position: relative;
            overflow: visible;
            height: 0px;
          }

          .creator-hover-region {
            position: relative;
            overflow: visible;
            top: -10px;
            height: 60px;
            text-align: center;
          }

          .cell-creator {
            display: none;
            background: var(--theme-cell-creator-bg);
            box-shadow: 0 1px 2px 0 rgba(0, 0, 0, 0.5);
            pointer-events: all;
            position: relative;
            top: -5px;
          }

          .cell-creator button {
            display: inline-block;

            width: 22px;
            height: 20px;
            padding: 0px 4px;

            text-align: center;

            border: none;
            outline: none;
            background: none;
          }

          .cell-creator button span {
            font-size: 15px;
            line-height: 1;

            color: var(--theme-cell-creator-fg);
          }

          .cell-creator button span:hover {
            color: var(--theme-cell-creator-fg-hover);
          }

          .creator-hover-region:hover > .cell-creator {
            display: inline-block;
          }

          .octicon {
            transition: color 0.5s;
          }
        `}</style>
      </div>
    );
  }
}

class CellCreator extends React.Component<ConnectedProps> {
  createCell = (type: "code" | "markdown"): void => {
    const {
      above,
      createCellBelow,
      createCellAppend,
      createCellAbove,
      id,
      contentRef
    } = this.props;

    if (id == null || typeof id != "string") {
      createCellAppend({ cellType: type, contentRef });
      return;
    }

    above
      ? createCellAbove({ cellType: type, id, contentRef })
      : createCellBelow({ cellType: type, id, source: "", contentRef });
  };

  render() {
    return (
      <PureCellCreator above={this.props.above} createCell={this.createCell} />
    );
  }
}

const mapDispatchToProps = dispatch => ({
  createCellAppend: (payload: *) => dispatch(actions.createCellAppend(payload)),
  createCellAbove: (payload: *) => dispatch(actions.createCellAbove(payload)),
  createCellBelow: (payload: *) => dispatch(actions.createCellBelow(payload))
});

// $FlowFixMe: react-redux typings
export default connect(
  null,
  mapDispatchToProps
)(CellCreator);
