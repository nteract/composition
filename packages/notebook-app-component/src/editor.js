// @flow
import { connect } from "react-redux";
import { actions, selectors } from "@nteract/core";
import type { ContentRef, AppState } from "@nteract/core";
import { omit } from "lodash";
import EditorView from "@nteract/editor";

type Props = {
  id: string,
  cellFocused: boolean,
  contentRef: ContentRef,
  options: Object
};

function mapStateToProps(state: AppState, ownProps: Props) {
  const kernel = selectors.currentKernel(state);
  const cursorBlinkRate = state.config.get("cursorBlinkRate", 530);
  return {
    // Don't propagate props only used for setting up dispatch and computed props
    ...omit(ownProps, [
      "id",
      "cellFocused",
      "contentRef",
      "options",
      "channels",
      "kernelStatus"
    ]),
    // Merge in our _one_ configurable options dealy
    // Ideally this would be Immutable or remain consistent so that
    // we're not thrashing updates and renders...
    options: Object.assign({}, ownProps.options, { cursorBlinkRate }),
    channels: kernel ? kernel.channels : null,
    kernelStatus: selectors.currentKernelStatus(state) || "not connected"
  };
}

const mapDispatchToProps = (dispatch, ownProps: Props) => {
  const { cellFocused, id, contentRef } = ownProps;

  return {
    onChange: (text: string) => {
      dispatch(actions.updateCellSource({ id, value: text, contentRef }));
    },

    onFocusChange(focused: boolean): void {
      if (focused) {
        dispatch(actions.focusCellEditor({ id, contentRef }));
        if (!cellFocused) {
          dispatch(actions.focusCell({ id, contentRef }));
        }
      }
    }
  };
};

// $FlowFixMe: react-redux typings
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(EditorView);
