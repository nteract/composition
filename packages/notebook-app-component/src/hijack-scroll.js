// @flow
/* eslint jsx-a11y/no-static-element-interactions: 0 */
/* eslint jsx-a11y/click-events-have-key-events: 0 */

import * as React from "react";

type HijackScrollProps = {
  focused: boolean,
  onClick: () => void,
  children: React.Node
};

export class HijackScroll extends React.Component<HijackScrollProps, *> {
  el: ?HTMLElement;

  scrollIntoViewIfNeeded(prevFocused?: boolean): void {
    // Check if the element is being hovered over.
    const hovered =
      this.el &&
      this.el.parentElement &&
      this.el.parentElement.querySelector(":hover") === this.el;

    if (
      this.props.focused &&
      prevFocused !== this.props.focused &&
      // Don't scroll into view if already hovered over, this prevents
      // accidentally selecting text within the codemirror area
      !hovered
    ) {
      if (this.el && "scrollIntoViewIfNeeded" in this.el) {
        // $FlowFixMe: This is only valid in Chrome, WebKit
        this.el.scrollIntoViewIfNeeded();
      } else {
        // TODO: Polyfill as best we can for the webapp version
      }
    }
  }

  componentDidUpdate(prevProps: HijackScrollProps) {
    this.scrollIntoViewIfNeeded(prevProps.focused);
  }

  componentDidMount(): void {
    this.scrollIntoViewIfNeeded();
  }

  render() {
    return (
      <div
        onClick={this.props.onClick}
        role="presentation"
        ref={el => {
          this.el = el;
        }}
      >
        {this.props.children}
      </div>
    );
  }
}
