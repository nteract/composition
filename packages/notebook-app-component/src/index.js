// @flow
import { hot } from "react-hot-loader";

import NotebookApp from "./notebook-app";

export default hot(module)(NotebookApp);
