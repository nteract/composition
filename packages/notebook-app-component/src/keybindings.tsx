import * as React
import { 
  Hotkey, 
  Hotkeys
} from "@blueprintjs/core";
