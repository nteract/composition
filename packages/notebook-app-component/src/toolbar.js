// @flow
/* eslint jsx-a11y/no-static-element-interactions: 0 */
/* eslint jsx-a11y/click-events-have-key-events: 0 */
/* eslint jsx-a11y/anchor-is-valid: 0 */

// TODO: Fix up a11y eslint here
// TODO: All the `<li>` below that have role button should just be `<button>` with proper styling

import * as React from "react";
import { connect } from "react-redux";
import { actions } from "@nteract/core";
import type { ContentRef } from "@nteract/core";
import {
  DropdownMenu,
  DropdownTrigger,
  DropdownContent
} from "@nteract/dropdown-menu";
import {
  TrashOcticon,
  ChevronDownOcticon,
  TriangleRightOcticon
} from "@nteract/octicons";

export type PureToolbarProps = {|
  type: "markdown" | "code" | "raw",
  executeCell: ?() => void,
  deleteCell: ?() => void,
  clearOutputs: ?() => void,
  toggleParameterCell: ?() => void,
  toggleCellInputVisibility: ?() => void,
  toggleCellOutputVisibility: ?() => void,
  toggleOutputExpansion: ?() => void,
  changeCellType: ?() => void,
  sourceHidden: boolean,
  contentRef: ContentRef
|};

export class PureToolbar extends React.Component<PureToolbarProps> {
  static defaultProps = {
    type: "code"
  };

  render(): React$Element<any> {
    const { type, executeCell, deleteCell, sourceHidden } = this.props;

    return (
      <div className="cell-toolbar-mask">
        <div className="cell-toolbar">
          {type !== "markdown" && (
            <button
              onClick={executeCell}
              title="execute cell"
              className="executeButton"
            >
              <span className="octicon">
                <TriangleRightOcticon />
              </span>
            </button>
          )}
          <DropdownMenu>
            <DropdownTrigger>
              <button title="show additional actions">
                <span className="octicon toggle-menu">
                  <ChevronDownOcticon />
                </span>
              </button>
            </DropdownTrigger>
            {type === "code" ? (
              <DropdownContent>
                <li
                  onClick={this.props.clearOutputs}
                  className="clearOutput"
                  role="option"
                  aria-selected="false"
                  tabIndex="0"
                >
                  <a>Clear Cell Output</a>
                </li>
                <li
                  onClick={this.props.toggleCellInputVisibility}
                  className="inputVisibility"
                  role="option"
                  aria-selected="false"
                  tabIndex="0"
                >
                  <a>Toggle Input Visibility</a>
                </li>
                <li
                  onClick={this.props.toggleCellOutputVisibility}
                  className="outputVisibility"
                  role="option"
                  aria-selected="false"
                  tabIndex="0"
                >
                  <a>Toggle Output Visibility</a>
                </li>
                <li
                  onClick={this.props.toggleOutputExpansion}
                  className="outputExpanded"
                  role="option"
                  aria-selected="false"
                  tabIndex="0"
                >
                  <a>Toggle Expanded Output</a>
                </li>
                <li
                  onClick={this.props.toggleParameterCell}
                  role="option"
                  aria-selected="false"
                  tabIndex="0"
                >
                  <a>Toggle Parameter Cell</a>
                </li>

                <li
                  onClick={this.props.changeCellType}
                  className="changeType"
                  role="option"
                  aria-selected="false"
                  tabIndex="0"
                >
                  <a>Convert to Markdown Cell</a>
                </li>
              </DropdownContent>
            ) : (
              <DropdownContent>
                <li
                  onClick={this.props.changeCellType}
                  className="changeType"
                  role="option"
                  aria-selected="false"
                  tabIndex="0"
                >
                  <a>Convert to Code Cell</a>
                </li>
              </DropdownContent>
            )}
          </DropdownMenu>
          <span className="spacer" />
          <button
            onClick={deleteCell}
            title="delete cell"
            className="deleteButton"
          >
            <span className="octicon">
              <TrashOcticon />
            </span>
          </button>
        </div>

        <style jsx>{`
          .cell-toolbar > div {
            display: inline-block;
          }

          .cell-toolbar {
            background-color: var(--theme-cell-toolbar-bg);
            opacity: 0.4;
            transition: opacity 0.4s;
          }

          .cell-toolbar:hover {
            opacity: 1;
          }

          .cell-toolbar button {
            display: inline-block;

            width: 22px;
            height: 20px;
            padding: 0px 4px;

            text-align: center;

            border: none;
            outline: none;
            background: none;
          }

          .cell-toolbar span {
            font-size: 15px;
            line-height: 1;
            color: var(--theme-cell-toolbar-fg);
          }

          .cell-toolbar button span:hover {
            color: var(--theme-cell-toolbar-fg-hover);
          }

          .cell-toolbar-mask {
            z-index: 9999;
            display: ${sourceHidden ? "block" : "none"};
            position: absolute;
            top: 0px;
            right: 0px;
            height: 34px;

            /* Set the left padding to 50px to give users extra room to move their
              mouse to the toolbar without causing the cell to go out of focus and thus
              hide the toolbar before they get there. */
            padding: 0px 0px 0px 50px;
          }

          .octicon {
            transition: color 0.5s;
          }

          .cell-toolbar span.spacer {
            display: inline-block;
            vertical-align: middle;
            margin: 1px 5px 3px 5px;
            height: 11px;
          }
        `}</style>
      </div>
    );
  }
}

// eslint-disable-next-line no-unused-vars
type ConnectedProps = {
  id: string,
  type: "markdown" | "code" | "raw",
  executeCell: () => void,
  deleteCell: () => void,
  clearOutputs: () => void,
  toggleCellOutputVisibility: () => void,
  toggleCellInputVisibility: () => void,
  changeCellType: () => void,
  toggleOutputExpansion: () => void
};

const mapDispatchToProps = (dispatch, { id, type, contentRef }) => ({
  toggleParameterCell: () =>
    dispatch(actions.toggleParameterCell({ id, contentRef })),
  deleteCell: () => dispatch(actions.deleteCell({ id, contentRef })),
  executeCell: () => dispatch(actions.executeCell({ id, contentRef })),
  clearOutputs: () => dispatch(actions.clearOutputs({ id, contentRef })),
  toggleCellOutputVisibility: () =>
    dispatch(actions.toggleCellOutputVisibility({ id, contentRef })),
  toggleCellInputVisibility: () =>
    dispatch(actions.toggleCellInputVisibility({ id, contentRef })),
  changeCellType: () =>
    dispatch(
      actions.changeCellType({
        id,
        to: type === "markdown" ? "code" : "markdown",
        contentRef
      })
    ),
  toggleOutputExpansion: () =>
    dispatch(actions.toggleOutputExpansion({ id, contentRef }))
});

// TODO: This toolbar could easily make use of ownProps (contentRef, cellId)
//       and pluck exactly the state it wants
// $FlowFixMe: react-redux typings
export default connect(
  null,
  mapDispatchToProps
)(PureToolbar);
