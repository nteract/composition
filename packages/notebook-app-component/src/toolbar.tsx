/* eslint jsx-a11y/no-static-element-interactions: 0 */
/* eslint jsx-a11y/click-events-have-key-events: 0 */
/* eslint jsx-a11y/anchor-is-valid: 0 */

// TODO: Fix up a11y eslint here
// TODO: All the `<li>` below that have role button should just be `<button>` with proper styling

import * as actions from "@nteract/actions";
import {
  DropdownContent,
  DropdownMenu,
  DropdownTrigger
} from "@nteract/dropdown-menu";
import {
  ChevronDownOcticon,
  TrashOcticon,
  TriangleRightOcticon
} from "@nteract/octicons";
import { ContentRef } from "@nteract/types";
import * as React from "react";
import { connect } from "react-redux";
import { Dispatch } from "redux";

import styled, { StyledComponent } from "styled-components";

export interface PureToolbarProps {
  type: "markdown" | "code" | "raw";
  executeCell?: () => void;
  deleteCell?: () => void;
  clearOutputs?: () => void;
  toggleParameterCell?: () => void;
  toggleCellInputVisibility?: () => void;
  toggleCellOutputVisibility?: () => void;
  toggleOutputExpansion?: () => void;
  changeCellType?: () => void;
  sourceHidden: boolean;
  contentRef: ContentRef;
}

export const CellToolbar = styled.div`
  background-color: var(--theme-cell-toolbar-bg);
  opacity: 0.4;
  transition: opacity 0.4s;

  & > div {
    display: inline-block;
  }

  :hover {
    opacity: 1;
  }

  button {
    display: inline-block;

    width: 22px;
    height: 20px;
    padding: 0px 4px;

    text-align: center;

    border: none;
    outline: none;
    background: none;
  }

  span {
    font-size: 15px;
    line-height: 1;
    color: var(--theme-cell-toolbar-fg);
  }

  button span:hover {
    color: var(--theme-cell-toolbar-fg-hover);
  }

  .octicon {
    transition: color 0.5s;
  }

  span.spacer {
    display: inline-block;
    vertical-align: middle;
    margin: 1px 5px 3px 5px;
    height: 11px;
  }
`;

export const CellToolbarMask = styled.div.attrs(
  (props: { sourceHidden?: boolean }) => ({
    style: {
      display: !props.sourceHidden ? "block" : "none"
    }
  })
)`
  z-index: 9999;
  position: absolute;
  top: 0px;
  right: 0px;
  height: 34px;

  /* Set the left padding to 50px to give users extra room to move their
              mouse to the toolbar without causing the cell to go out of focus and thus
              hide the toolbar before they get there. */
  padding: 0px 0px 0px 50px;
` as StyledComponent<"div", any, { sourceHidden?: boolean }, never>;

export class PureToolbar extends React.PureComponent<PureToolbarProps> {
  static defaultProps: Partial<PureToolbarProps> = {
    type: "code"
  };

  render() {
    const { type, executeCell, deleteCell, sourceHidden } = this.props;

    return (
      <CellToolbarMask sourceHidden={sourceHidden}>
        <CellToolbar>
          {type !== "markdown" && (
            <button
              onClick={executeCell}
              title="execute cell"
              className="executeButton"
            >
              <span className="octicon">
                <TriangleRightOcticon />
              </span>
            </button>
          )}
          <DropdownMenu>
            <DropdownTrigger>
              <button title="show additional actions">
                <span className="octicon toggle-menu">
                  <ChevronDownOcticon />
                </span>
              </button>
            </DropdownTrigger>
            {type === "code" ? (
              <DropdownContent>
                <li
                  onClick={this.props.clearOutputs}
                  className="clearOutput"
                  role="option"
                  aria-selected="false"
                  tabIndex={0}
                >
                  <a>Clear Cell Output</a>
                </li>
                <li
                  onClick={this.props.toggleCellInputVisibility}
                  className="inputVisibility"
                  role="option"
                  aria-selected="false"
                  tabIndex={0}
                >
                  <a>Toggle Input Visibility</a>
                </li>
                <li
                  onClick={this.props.toggleCellOutputVisibility}
                  className="outputVisibility"
                  role="option"
                  aria-selected="false"
                  tabIndex={0}
                >
                  <a>Toggle Output Visibility</a>
                </li>
                <li
                  onClick={this.props.toggleOutputExpansion}
                  className="outputExpanded"
                  role="option"
                  aria-selected="false"
                  tabIndex={0}
                >
                  <a>Toggle Expanded Output</a>
                </li>
                <li
                  onClick={this.props.toggleParameterCell}
                  role="option"
                  aria-selected="false"
                  tabIndex={0}
                >
                  <a>Toggle Parameter Cell</a>
                </li>

                <li
                  onClick={this.props.changeCellType}
                  className="changeType"
                  role="option"
                  aria-selected="false"
                  tabIndex={0}
                >
                  <a>Convert to Markdown Cell</a>
                </li>
              </DropdownContent>
            ) : (
              <DropdownContent>
                <li
                  onClick={this.props.changeCellType}
                  className="changeType"
                  role="option"
                  aria-selected="false"
                  tabIndex={0}
                >
                  <a>Convert to Code Cell</a>
                </li>
              </DropdownContent>
            )}
          </DropdownMenu>
          <span className="spacer" />
          <button
            onClick={deleteCell}
            title="delete cell"
            className="deleteButton"
          >
            <span className="octicon">
              <TrashOcticon />
            </span>
          </button>
        </CellToolbar>
      </CellToolbarMask>
    );
  }
}

interface ConnectedProps {
  id: string;
  type: "markdown" | "code" | "raw";
  executeCell: () => void;
  deleteCell: () => void;
  clearOutputs: () => void;
  toggleCellOutputVisibility: () => void;
  toggleCellInputVisibility: () => void;
  changeCellType: () => void;
  toggleOutputExpansion: () => void;
}

interface InitialProps {
  id: string;
  type: "markdown" | "code" | "raw";
  contentRef: ContentRef;
}

const makeMapDispatchToProps = (
  initialDispatch: Dispatch,
  initialProps: InitialProps
) => {
  const { contentRef, id, type } = initialProps;
  const mapDispatchToProps = (dispatch: Dispatch) => {
    return {
      changeCellType: () =>
        dispatch(
          actions.changeCellType({
            contentRef,
            id,
            to: type === "markdown" ? "code" : "markdown"
          })
        ),
      clearOutputs: () => dispatch(actions.clearOutputs({ id, contentRef })),
      deleteCell: () => dispatch(actions.deleteCell({ id, contentRef })),
      executeCell: () => dispatch(actions.executeCell({ id, contentRef })),
      toggleCellInputVisibility: () =>
        dispatch(actions.toggleCellInputVisibility({ id, contentRef })),
      toggleCellOutputVisibility: () =>
        dispatch(actions.toggleCellOutputVisibility({ id, contentRef })),
      toggleOutputExpansion: () =>
        dispatch(actions.toggleOutputExpansion({ id, contentRef })),
      toggleParameterCell: () =>
        dispatch(actions.toggleParameterCell({ id, contentRef }))
    };
  };
  return mapDispatchToProps;
};

// TODO: This toolbar could easily make use of ownProps (contentRef, cellId)
//       and pluck exactly the state it wants
export default connect(
  null,
  makeMapDispatchToProps
)(PureToolbar);
