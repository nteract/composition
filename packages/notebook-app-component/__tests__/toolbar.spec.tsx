import { fixtureStore } from "@nteract/fixtures";
import { ContentRecord, ContentRef } from "@nteract/types";
import { mount } from "enzyme";
import toJSON from "enzyme-to-json";
import * as Immutable from "immutable";
import * as React from "react";
import { Provider } from "react-redux";

import { CellToolbar } from "../src/toolbar";

const getContentFromStore = (store) => {
  const contents: Immutable.Map<ContentRef, ContentRecord> =
    store.getState().core.entities.contents.byRef;
  const contentRef = contents.keys().next().value;
  const contentRecord = contents.values().next().value.toJS();
  const cellID = contentRecord.model.notebook.cellOrder[0];
  const cellRecord = contentRecord.model.notebook.cellMap[cellID];

  return {
    contentRef,
    contentRecord,
    cellID,
    cellRecord,
  };
};

const makeToolbar = () => {
  const store = fixtureStore({codeCellCount: 1});
  const { contentRef, cellID } = getContentFromStore(store);
  const toolbar = mount(
    <Provider store={store}>
      <CellToolbar
        isCellFocused
        isSourceHidden={false}
        contentRef={contentRef}
        id={cellID}
      />
    </Provider>
  );

  return {
    toolbar,
    store,
  };
};

describe("Toolbar", () => {
  test("should be able to render a toolbar", () => {
    const { toolbar } = makeToolbar();
    expect(toJSON(toolbar.render())).toMatchSnapshot();
    toolbar.find(".toolbar-dropdown button").simulate("click");
    expect(toJSON(toolbar.render())).toMatchSnapshot();
  });

  test("can change cell type", () => {
    const { toolbar, store } = makeToolbar();
    const { cellRecord: cellPre } = getContentFromStore(store);
    toolbar.find(".toolbar-dropdown button").simulate("click");
    toolbar.find(".change-to-markdown").simulate("click");
    const { cellRecord: cellPost } = getContentFromStore(store);
    expect(cellPre.cell_type).toBe("code");
    expect(cellPost.cell_type).toBe("markdown");
  });

  test("can delete cell", () => {
    const { toolbar, store } = makeToolbar();
    const { cellRecord: cellPre } = getContentFromStore(store);
    toolbar.find(".delete-cell").simulate("click");
    const { cellRecord: cellPost } = getContentFromStore(store);
    expect(cellPre).toBeDefined();
    expect(cellPost).toBeUndefined();
  });
});
