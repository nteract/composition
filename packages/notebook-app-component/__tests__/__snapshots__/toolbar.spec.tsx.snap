// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar should be able to render a toolbar 1`] = `
<div
  class="sc-ifAKCX fYqCVQ"
>
  <div
    class="toolbar"
  >
    <div
      class="sc-bxivhb eccFiY"
    >
      <button
        class="run-cell icon"
        role="menuitem"
      >
        <span>
          <svg
            height="16"
            style="fill: currentColor; display: inline-block; vertical-align: text-bottom;"
            viewBox="0 0 6 16"
            width="6"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              Run Cell
            </title>
            <path
              d="M0 14l6-6-6-6z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
  <div
    class="toolbar-dropdown"
  >
    <div
      class="sc-bdVaJa igpjjl"
    >
      <div
        class="sc-bwzfXH iaRngl"
      >
        <button>
          <span>
            <svg
              height="16"
              style="fill: currentColor; display: inline-block; vertical-align: text-bottom;"
              viewBox="0 0 10 16"
              width="10"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                Show Additional Actions
              </title>
              <path
                d="M5 11L0 6l1.5-1.5L5 8.25 8.5 4.5 10 6z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    class="toolbar-end"
  >
    <div
      class="sc-bxivhb eccFiY"
    >
      <button
        class="delete-cell icon"
        role="menuitem"
      >
        <span>
          <svg
            height="16"
            style="fill: currentColor; display: inline-block; vertical-align: text-bottom;"
            viewBox="0 0 12 16"
            width="12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              Delete Cell
            </title>
            <path
              d="M11 2H9c0-.55-.45-1-1-1H5c-.55 0-1 .45-1 1H2c-.55 0-1 .45-1 1v1c0 .55.45 1 1 1v9c0 .55.45 1 1 1h7c.55 0 1-.45 1-1V5c.55 0 1-.45 1-1V3c0-.55-.45-1-1-1zm-1 12H3V5h1v8h1V5h1v8h1V5h1v8h1V5h1v9zm1-10H2V3h9v1z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Toolbar should be able to render a toolbar 2`] = `
<div
  class="sc-ifAKCX fYqCVQ"
>
  <div
    class="toolbar"
  >
    <div
      class="sc-bxivhb eccFiY"
    >
      <button
        class="run-cell icon"
        role="menuitem"
      >
        <span>
          <svg
            height="16"
            style="fill: currentColor; display: inline-block; vertical-align: text-bottom;"
            viewBox="0 0 6 16"
            width="6"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              Run Cell
            </title>
            <path
              d="M0 14l6-6-6-6z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
  <div
    class="toolbar-dropdown"
  >
    <div
      class="sc-bdVaJa igpjjl"
    >
      <div
        class="sc-bwzfXH iaRngl"
      >
        <button>
          <span>
            <svg
              height="16"
              style="fill: currentColor; display: inline-block; vertical-align: text-bottom;"
              viewBox="0 0 10 16"
              width="10"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                Show Additional Actions
              </title>
              <path
                d="M5 11L0 6l1.5-1.5L5 8.25 8.5 4.5 10 6z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </button>
      </div>
      <div
        class="sc-htpNat juMQnP"
      >
        <ul
          aria-label="dropdown-content"
          role="listbox"
        >
          <li
            role="presentation"
          >
            <div
              class="sc-bxivhb eccFiY"
            >
              <button
                class="clear-output text"
                disabled=""
                role="menuitem"
              >
                Clear Cell Output
              </button>
            </div>
          </li>
          <li
            role="presentation"
          >
            <div
              class="sc-bxivhb eccFiY"
            >
              <label
                aria-checked="true"
                class="toggle-input checked"
                role="menuitemcheckbox"
              >
                <input
                  aria-checked="true"
                  checked=""
                  role="menuitemcheckbox"
                  type="checkbox"
                />
                Show Cell Input
              </label>
            </div>
          </li>
          <li
            role="presentation"
          >
            <div
              class="sc-bxivhb eccFiY"
            >
              <label
                aria-checked="true"
                class="toggle-output checked"
                role="menuitemcheckbox"
              >
                <input
                  aria-checked="true"
                  checked=""
                  role="menuitemcheckbox"
                  type="checkbox"
                />
                Show Cell Output
              </label>
            </div>
          </li>
          <li
            role="presentation"
          >
            <div
              class="sc-bxivhb eccFiY"
            >
              <label
                aria-checked="false"
                class="toggle-output-expanded not-checked"
                role="menuitemcheckbox"
              >
                <input
                  aria-checked="false"
                  role="menuitemcheckbox"
                  type="checkbox"
                />
                Expanded Cell Output
              </label>
            </div>
          </li>
          <li
            role="presentation"
          >
            <div
              class="sc-bxivhb eccFiY"
            >
              <button
                class="change-to-markdown text"
                role="menuitem"
              >
                Convert to Markdown Cell
              </button>
            </div>
          </li>
          <li
            role="presentation"
          >
            <div
              class="sc-bxivhb eccFiY"
            >
              <label
                aria-checked="false"
                class="toggle-parameter not-checked"
                role="menuitemcheckbox"
              >
                <input
                  aria-checked="false"
                  role="menuitemcheckbox"
                  type="checkbox"
                />
                Is Parameter Cell
              </label>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    class="toolbar-end"
  >
    <div
      class="sc-bxivhb eccFiY"
    >
      <button
        class="delete-cell icon"
        role="menuitem"
      >
        <span>
          <svg
            height="16"
            style="fill: currentColor; display: inline-block; vertical-align: text-bottom;"
            viewBox="0 0 12 16"
            width="12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              Delete Cell
            </title>
            <path
              d="M11 2H9c0-.55-.45-1-1-1H5c-.55 0-1 .45-1 1H2c-.55 0-1 .45-1 1v1c0 .55.45 1 1 1v9c0 .55.45 1 1 1h7c.55 0 1-.45 1-1V5c.55 0 1-.45 1-1V3c0-.55-.45-1-1-1zm-1 12H3V5h1v8h1V5h1v8h1V5h1v8h1V5h1v9zm1-10H2V3h9v1z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</div>
`;
